lib/model/model.go

```diff
@@ -1499,7 +1499,9 @@ func (m *Model) updateLocals(folder string, fs []protocol.FileInfo) {
 
 func (m *Model) localChangeDetected(folder config.FolderConfiguration, files []protocol.FileInfo) {
 	// For windows paths, strip unwanted chars from the front
+	m.fmut.RLock()
```

Comment:
The locking is needed where you access m.folderCfgs, as that is a non-concurrent-safe map that may be changed by other routines in parallell.